python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -353,3 +353,95 @@ def _doConvolve(exposure, kernel):
         afwMath.convolve(outExp.getMaskedImage(), exposure.getMaskedImage(), kern, convCntrl)
 
         return outExp, kern
+
+
+class DecorrelateALKernelMapperTask(DecorrelateALKernelTask):
+    """Task to be used as an ImageMapperSubtask for computing
+    A&L decorrelation on subimages on a grid across a A&L difference image.
+
+    This task subclasses DecorrelateALKernelTask in order to implement
+    all of that task's configuration parameters, as well as its `run` method.
+    """
+    ConfigClass = DecorrelateALKernelConfig
+    _DefaultName = 'ip_diffim_decorrelateALKernelMapper'
+
+    def __init__(self, *args, **kwargs):
+        DecorrelateALKernelTask.__init__(self, *args, **kwargs)
+
+    def run(self, subExposure, expandedSubExposure, fullBBox, **kwargs):
+        """Perform decorrelation operation on `subExposure`, using
+        `expandedSubExposure` to allow for invalid edge pixels arising from
+        convolutions.
+
+        This method performs A&L decorrelation on `subExposure` using
+        local measures for image variances and PSF. `subExposure` is a
+        sub-exposure of the non-decorrelated A&L diffim. It also
+        requires the corresponding sub-exposures of the template
+        (`template`) and science (`science`) exposures.
+
+        Parameters
+        ----------
+        subExposure : afw.Exposure
+            the sub-exposure of the diffim
+        expandedSubExposure : afw.Exposure
+            the expanded sub-exposure upon which to operate
+        fullBBox : afwGeom.BoundingBox
+            the bounding box of the original exposure
+        template : afw.Exposure
+            the corresponding sub-exposure of the template exposure
+        science : afw.Exposure
+            the corresponding sub-exposure of the science exposure
+        alTaskResult : pipeBase.Struct
+            the result of A&L image differencing on `science` and
+            `template`, importantly containing the resulting
+            `psfMatchingKernel`. Can be `None`, only if
+            `psfMatchingKernel` is not `None`.
+        psfMatchingKernel : Alternative parameter for passing the
+            A&L `psfMatchingKernel` directly.
+        kwargs :
+            additional keyword arguments propagated from
+            `ImageMapReduceTask.run`.
+
+        Returns
+        -------
+        A `pipeBase.Struct containing the result of the `subExposure`
+        processing, labelled 'subExposure'. It also returns the
+        'decorrelationKernel', although that currently is not used.
+
+        Notes
+        -----
+        This `run` method accepts parameters identical to those of
+        `ImageMapperSubtask.run`, since it is called from the
+        `ImageMapperTask`.  See that class for more information.
+        """
+        logLevel = self.log.getLevel()
+        self.log.setLevel(lsst.log.ERROR)  # Prevent too much log verbosity from DecorrelateALKernelTask.run
```

Comment:
I'm a little worried that the questions raised by @djreiss will be lost in the noise in Jira.  I think it might be worth having a broader discussion.  Is a community post reasonable?